tests/testthat/test-with.R

```diff
@@ -14,20 +14,17 @@ test_that("linters_with_defaults warns on unused NULLs", {
 test_that("all default linters are tagged default", {
   expect_named(linters_with_defaults(), available_linters(tags = "default")$linter)
 
-  # TODO(michaelchirico): use plain expect_equal after waldo#133 makes it into a CRAN release
-  # Here, the environment()s are different because factories use them.
-  skip_if_not_r_version("4.1.0") # Desired all.equal behaviour only available in >= 4.1
+  skip_if_not_installed("waldo", "0.4.0") # needs waldo#133
   # covr modifies package functions causing differing deparse() results even for identical anonymous functions.
   # This happens because default_linters is generated at build time and thus not modifiable by covr, whereas
   # linters_with_tags() constructs the linters at runtime.
   skip_if(requireNamespace("covr", quietly = TRUE) && covr::in_covr())
 
-  # all.equal.default warns on some releases of R if the input is a function, see #1392
-  expect_silent(result <- all.equal(linters_with_tags("default"), linters_with_defaults()))
-  expect_true(result)
+  expect_equal(linters_with_tags("default"), linters_with_defaults())
```

Comment:
this no longer relies on `all.equal.default`, is that right? since our waldo & testthat versions are required to be recent enough?

Reply:
Not directly, no.

`all.equal()` is used just once in the following test, which is skipped if R version is below `4.1`:

``` r
library(sloop)
library(lintr)

s3_dispatch(all.equal(
  linters_with_tags("default"), 
  linters_with_defaults(line_length_linter(120L)))
)
#> => all.equal.list
#>  * all.equal.default
```

<sup>Created on 2022-09-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
:thinking: sorry, a bit confused. `expect_equal()` doesn't call `all.equal()` at all, right?

```r
debug(all.equal)
expect_equal(linters_with_tags("default"), linters_with_defaults()) # no browser launched
```

Reply:
Sorry for the confusion. 

The test in question uses `all.equal()` directly:

```r
  expect_equal(
    all.equal(linters_with_tags("default"), linters_with_defaults(line_length_linter(120L))),
    'Component "line_length_linter": Component "length": Mean relative difference: 0.5'
  )
```

Reply:
ohh i see, got it.